lib/Ethereum/RPC/Contract.pm

```diff
@@ -191,15 +195,18 @@ sub _prepare_transaction {
 
     my $data = $compiled_data . $encoded;
 
-    return Ethereum::RPC::Contract::ContractTransaction->new(
+    my $transaction = Ethereum::RPC::Contract::ContractTransaction->new(
         contract_address => $self->contract_address,
         rpc_client       => $self->rpc_client,
         data             => $self->append_prefix($data),
         from             => $self->from,
-        gas              => $self->gas,
-        gas_price        => $self->gas_price,
+        gas              => $self->gas
     );
 
+    $transaction->{gas_price}                = $self->gas_price                if $self->gas_price;
+    $transaction->{max_fee_per_gas}          = $self->max_fee_per_gas          if $self->max_fee_per_gas;
+    $transaction->{max_priority_fee_per_gas} = $self->max_priority_fee_per_gas if $self->max_priority_fee_per_gas;
```

Comment:
Curious to know what will be the outcome when all these iffs will be true for a Type 2 transaction?
should we just ignore the $transaction->{gas_price} case when any of max_fee_per_gas & max_priority_fee_per_gas are available?